### PR TITLE
Add mach command to update public domain list and use a HashSet for matching

### LINKED ISSUE #11216: Write a mach command to generate the contents of pub_domains.rs
It was committed as part of #4519 with no documentation as to how it was created. It should be documented, and have an easy way to regenerate it as the list of official TLDs gets updated in the future.

### PR DESCRIPTION
**Pull request by fduraffourg**
Add a `mach update-pub-domains` command that update the public domain list used by servo.
Replace the Vec matching with a HashSet which is way quicker.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #11216

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11513)

### CODE REVIEW

File: tests/unit/net/cookie_http_state.rs [hunk 1]
```diff
@@ -975,7 +975,6 @@ fn test_domain0016() {
 }
 
 #[test]
-#[should_panic] // Look at cookie_http_state_utils.py if this test fails
```
Comment (jdm) [inline tests/unit/net/cookie_http_state.rs]: We should update cookie_http_state_utils.py to adjust for this.

File: tests/unit/net/pub_domains.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    assert!(is_pub_domain("teledata.mz") == false);
+}
+
+#[test]
+fn test_is_pub_domain_not() {
+    assert!(is_pub_domain(".servo.org") == false);
+    assert!(is_pub_domain("www.mozilla.org") == false);
+    assert!(is_pub_domain("publicsuffix.org") == false);
+    assert!(is_pub_domain("hello.world.jm") == false);
+    assert!(is_pub_domain("toto.toto.kobe.jp") == false);
```
Comment (jdm) [inline tests/unit/net/pub_domains.rs]: Let's make the asserts using == be `assert_eq!(is_pub_domain("..."), false)` instead.

**Comment by jdm** [comment]
This looks really good! Apologies about the delay.

**Comment by jdm** [comment]
@bors-servo: retry
* weird SSL failure?

**Comment by jdm** [comment]
@bors-servo: retry
* #11682

**Comment by jdm** [comment]
@bors-servo: retry
* #11561

**Comment by jdm** [comment]
@bors-servo: retry
* #11682

**Comment by shinglyu** [comment]
@fduraffourg Thank you for this patch. Our performance testing result shows that this improves the performance of page loading time by ~25%! 

selection_012

**Comment by jdm** [comment]
Turns out that a hashmap is significantly faster than comparing against a huge list of strings...

**Comment by fduraffourg** [comment]
Cool! :)

**Pull request merged**